File: 프로그래머스/Lv.2/43165. 타겟 넘버/타겟 넘버.py
def solution(numbers, target):
    global answer
    answer = 0
    def dfs(cnt, sum_num):
        global answer
        if cnt == len(numbers):
            if sum_num == target:
                answer += 1
            return answer
        dfs(cnt+1, sum_num+numbers[cnt])
        dfs(cnt+1, sum_num-numbers[cnt])
    dfs(0, 0)
    return answer

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.java
class Solution {
    static int answer = 0;
    public int solution(int[] numbers, int target) {
        dfs(1, numbers, target, numbers[0]);
        dfs(1, numbers, target, -numbers[0]);
        return answer;
    }
    
    static void dfs(int len, int[] numbers, int target, int num) {
        if (len == numbers.length) {
            if (num == target) answer++;
            return;
        }
        dfs(len+1, numbers, target, num+numbers[len]);
        dfs(len+1, numbers, target, num-numbers[len]);
    }
}